Console/UpgradeHelperCommand.php

```diff
@@ -30,7 +31,25 @@ protected function configure()
     protected function execute(InputInterface $input, OutputInterface $output)
     {
         $diff = file($input->getArgument('diff'));
-        $result = $this->runner->run($diff);
+        $lines = count($diff);
+
+        $result = [];
+        $result['preferences'] = [];
+        $result['overrides'] = [];
+
+        $progressBar = new ProgressBar($output, $lines);
+        $progressBar->setFormat('Lines in diff processed: %current%/%max% [%bar%] %percent:3s%% (elapsed: %elapsed% | remaining: ~%remaining%)');
+
+        foreach ($diff as $line) {
+            $progressBar->advance();
+            extract($this->runner->run($line));
```

Comment:
Another thought. For readability/long term maintenance do we want to comment here on extract? 
```
/* 
* Extract will populate: 
* $type
* $path
* $items
* /
```
Essentially keep track of the intended variables extract is meant to define here.

Reply:
Yeah...extract is weird and magical in a way that I don't love, but it was appropriate here.